portal/src/graphql/adminapi/UserDetailsConnectedIdentities.tsx

```diff
@@ -49,39 +52,41 @@ interface UserDetailsConnectedIdentitiesProps {
   availableLoginIdIdentities: string[];
 }
 
-const identityTypes = ["email", "phone", "username"] as const;
-type IdentityType = typeof identityTypes[number];
+const loginIdIdentityTypes = ["email", "phone", "username"] as const;
+type LoginIDIdentityType = typeof loginIdIdentityTypes[number];
+type IdentityType = "login_id" | "oauth";
 
-interface EmailIdentityListItem {
+interface OAuthIdentityListItem {
   id: string;
-  email: string;
+  type: "oauth";
+  providerType: OAuthSSOProviderType;
+  name: string;
   verified: boolean;
   connectedOn: string;
 }
 
-interface PhoneIdentityListItem {
+interface LoginIDIdentityListItem {
   id: string;
-  phone: string;
-  verified: boolean;
-  addedOn: string;
-}
-
-interface UsernameIdentityListItem {
-  id: string;
-  username: string;
+  type: "login_id";
+  key: "email" | "phone" | "username";
+  value: string;
+  verified?: boolean;
   addedOn: string;
 }
 
 export interface IdentityLists {
-  email: EmailIdentityListItem[];
-  phone: PhoneIdentityListItem[];
-  username: UsernameIdentityListItem[];
+  oauth: OAuthIdentityListItem[];
+  email: LoginIDIdentityListItem[];
+  phone: LoginIDIdentityListItem[];
+  username: LoginIDIdentityListItem[];
 }
 
 interface IdentityListCellProps {
   identityID: string;
   identityType: IdentityType;
-  identityName: string;
+  loginIdIdentityType?: LoginIDIdentityType;
```

Comment:
I'd like to combine the fields `loginIdIdentityType`, `oauthProviderType`, and `identityName`.

These fields are used to represent mutual exclusive structure: if `identityType` is OAuth, `oauthProviderType` presents and `loginIdIdentityType` absents, and same for `identityName`. We can use a union type `OAuthIdentityListItem | LoginIDIdentityListItem` to represent all necessary identity data, and extract view data through helper functions:
```
function getIcon(item: OAuthIdentityListItem | LoginIDIdentityListItem): React.ReactNode;
function getName(item: OAuthIdentityListItem | LoginIDIdentityListItem): string | undefined;
```

Reply:
I see, pushed 1 more commit